rosco-web/src/main/groovy/com/netflix/spinnaker/rosco/controllers/BakeryController.groovy

```diff
@@ -241,7 +241,16 @@ class BakeryController {
     def bake = bakeStore.retrieveBakeDetailsById(bakeId)
 
     if (bake) {
-      return bake
+      if (bake.artifacts?.size() > 0) {
+        bake.artifact = bake.artifacts.find{ it.location == region}
+        bake.regions = bake.artifacts.collect{it.location}
+        bake.ami = bake.artifact.reference
+        bake.artifacts = null
+
+        return bake
+      } else {
+        return bake
+      }
```

Comment:
Drop the `else` and just always return bake

Reply:
fixed